DSCResources/cChocoSource/cChocoSource.psm1

```diff
@@ -0,0 +1,160 @@
+function Get-TargetResource
+{
+    [CmdletBinding()] 
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,   
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure='Present',
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [UInt32]
+        $Priority,
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Source
+    )
+
+    Write-Verbose "Start Get-TargetResource"
+
+    #Needs to return a hashtable that returns the current
+    #status of the configuration component
+    $Configuration = @{
+        Name = $Name
+        Priority = $Priority
+        Source = $Source
+    }
+
+    return $Configuration
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]    
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Name,   
+        [ValidateSet('Present','Absent')]
+        [System.String]
+        $Ensure='Present',
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [UInt32]
+        $Priority,
+        [parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Source
+    )
+    Write-Verbose "Start Set-TargetResource"
+
+	if($Ensure -eq "Present")
+	{
+		if($priority -eq $null)
+		{
+			choco sources add -n"$name" -s"$source"
+		}
+		else
+		{
+			choco sources add -n"$name" -s"$source" --priority=$priority
+		}
```

Comment:
Passwords in DSC resources should always be PSCredential objects.  That's the only thing that will trigger encryption of the password in the MOF file (if you've set up certificates for the node).

Inside the resource, you can fetch the plain-text version of the password easily with `$Credential.GetNetworkCredential().Password`